quesma/quesma/search.go

```diff
@@ -213,129 +376,110 @@ func (q *QueryRunner) handleSearchCommon(ctx context.Context, indexPattern strin
+	}
 
-		plan.Queries, err = registry.QueryTransformerFor(table.Name, q.cfg, q.schemaRegistry).Transform(plan.Queries)
-		if err != nil {
-			logger.ErrorWithCtx(ctx).Msgf("error transforming queries: %v", err)
-		}
-
-		if canParse {
-			queries := plan.Queries
-			if len(queries) > 0 && query_util.IsNonAggregationQuery(queries[0]) {
-				if properties := q.findNonexistingProperties(queries[0], table, queryTranslator); len(properties) > 0 {
-					logger.DebugWithCtx(ctx).Msgf("properties %s not found in table %s", properties, table.Name)
-					if elasticsearch.IsIndexPattern(indexPattern) {
-						return queryparser.EmptySearchResponse(ctx), nil
-					} else {
-						return nil, fmt.Errorf("properties %s not found in table %s", properties, table.Name)
-					}
-				}
-			}
-			go func() {
-				defer recovery.LogAndHandlePanic(ctx, func(err error) {
-					doneCh <- AsyncSearchWithError{err: err}
-				})
-
-				translatedQueryBody, results, err := q.searchWorker(ctx, plan, table, doneCh, optAsync)
-				if err != nil {
-					doneCh <- AsyncSearchWithError{err: err}
-					return
-				}
+	plan.IndexPattern = indexPattern
+	plan.StartTime = startTime
+	plan.Name = model.MainExecutionPlan
 
-				if len(results) == 0 {
-					logger.ErrorWithCtx(ctx).Msgf("no hits, sqls: %v", translatedQueryBody)
-					doneCh <- AsyncSearchWithError{translatedQueryBody: translatedQueryBody, err: errors.New("no hits")}
-					return
-				}
+	var alternativePlan *model.ExecutionPlan
 
-				results, err = q.postProcessResults(table, results)
-				if err != nil {
-					doneCh <- AsyncSearchWithError{translatedQueryBody: translatedQueryBody, err: err}
-				}
+	// TODO add alternative plan here
 
-				if plan.ResultAdapter != nil {
-					results, err = plan.ResultAdapter.Transform(results)
-					if err != nil {
-						doneCh <- AsyncSearchWithError{translatedQueryBody: translatedQueryBody, err: err}
-					}
-				}
+	/* You may use this code to run alternative plan for checking how it works
+	   It breaks the tests. So, it is commented out.
 
-				searchResponse := queryTranslator.MakeSearchResponse(plan.Queries, results)
+	alternativePlan = &model.ExecutionPlan{
+		IndexPattern:          plan.IndexPattern,
+		QueryRowsTransformers: plan.QueryRowsTransformers,
+		ResultAdapter:         plan.ResultAdapter,
+		Queries:               plan.Queries,
+		StartTime:             plan.StartTime,
+		Name:                  model.AlternativeExecutionPlan,
+	}
+	*/
 
-				doneCh <- AsyncSearchWithError{response: searchResponse, translatedQueryBody: translatedQueryBody, err: err}
-			}()
+	var executionChan chan executionPlanResult
 
-		} else {
-			queries := plan.Queries
-			queriesBody := make([]types.TranslatedSQLQuery, len(queries))
-			queriesBodyConcat := ""
-			for i, query := range queries {
-				queriesBody[i].Query = []byte(query.SelectCommand.String())
-				queriesBodyConcat += query.SelectCommand.String() + "\n"
+	if alternativePlan != nil {
```

Comment:
Add comment, that we run alternative plan, I would even move it to different method.

Reply:
I've extracted a method. It looks clean now.